File: server/src/main/java/com/smartexpense/api/application/port/in/AuthUseCase.java
package com.smartexpense.api.application.port.in;

import com.smartexpense.api.application.dto.request.LoginRequestDto;
import com.smartexpense.api.application.dto.request.RegisterRequestDto;
import com.smartexpense.api.application.dto.response.AuthResponseDto;

public interface AuthUseCase {
    AuthResponseDto login(LoginRequestDto request);
    AuthResponseDto register(RegisterRequestDto request);
}

File: server/src/main/kotlin/com/smartexpense/api/application/port/in/AuthUseCase.kt
package com.smartexpense.api.application.port.`in`

import com.smartexpense.api.application.dto.request.LoginRequestDto
import com.smartexpense.api.application.dto.request.RegisterRequestDto
import com.smartexpense.api.application.dto.response.AuthResponseDto

interface AuthUseCase {
    fun login(request: LoginRequestDto): AuthResponseDto
    fun register(request: RegisterRequestDto): AuthResponseDto
}
